### ISSUE TITLE
Linear Regression

### ISSUE THREAD

**Issue by arturfernan** [arturfernan/DataScience#1]
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import Ridge
from sklearn.metrics import r2_score
import matplotlib.pyplot as plt

# Extrair dataset 
df = pd.read_csv('/kaggle/input/marketing-linear-multiple-regression/Marketing_Data.csv')

# Definindo variáveis 
X = df[['youtube', 'facebook']].values
Y = df['sales'].values

# Criar dados de treino e teste 
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.30, random_state=42)

# Criando Regressão Ridge e treinando o modelo
regressor = Ridge(alpha=1.0)  # Você pode ajustar o valor de alpha para regularização mais forte ou mais fraca
regressor.fit(X_train, Y_train)

# Criando predição de dados 
y_pred = regressor.predict(X_test)

# Medindo acurácia do modelo 
score = r2_score(Y_test, y_pred)
print(f'A acurácia do modelo Ridge é de: {score:.2f}.')

# Criando equação de predição 
intercept = regressor.intercept_
coefs = regressor.coef_
terms = ['YouTube', 'Facebook']

# Exibindo os coeficientes e o intercepto de maneira mais clara
print("\nEquação da Regressão Ridge:")

for coef, term in zip(coefs, terms):
    print(f'{term}: {coef:.2f} (impacto nas vendas por unidade adicional de investimento)')
    
# Exibindo o intercepto
print(f"Intercepto (vendas previstas sem investimento): {intercept:.2f}")

# Exibindo a equação final do modelo de maneira interpretativa
print("\nInterpretação do Modelo:")
print(f"Vendas previstas (sales) = {intercept:.2f} + {coefs[0]:.2f} * YouTube + {coefs[1]:.2f} * Facebook")

# Plotando gráfico de vendas reais vs vendas previstas
plt.figure(figsize=(10, 6))
plt.scatter(Y_test, y_pred, color='blue', label='Predições')
plt.plot([min(Y_test), max(Y_test)], [min(Y_test), max(Y_test)], color='red', lw=2, label='Linha Ideal')
plt.xlabel('Vendas Reais')
plt.ylabel('Vendas Previstas')
plt.title('Vendas Reais vs. Vendas Previstas (Regressão Ridge)')
plt.legend()
plt.grid(True)
plt.show()
